### PR TITLE
Prealpha setup work 1

### PR DESCRIPTION
**Pull request by LT-Rascek**
### Character History
- [x] Add BM476 Foederati, Tributary for those powers with such relationships:
   * ERE: Thracian Goths (Foederati), Lazica (Tributary), Sarir (Tributary), Ghassanids (Foederati)
   * Persia: Iberia (Tributary), Caucasian Albania (Tributary), Lakhmids (Tributary), Bukhara (Tributary), Khwarezem (Tributary)
   * Hephthalites (all Tributaries): Khuttal, Samarkand, Chach, Oshrusana, Fergana, Dahae, Zunbils, Kashgar

### Vanilla Feature Suppression
- [x] Remove ahistorical vanilla decisions
- [x] Remove ahistorical vanilla cultures
- [x] Remove ahistorical vanilla faiths
- [x] Remove ahistorical vanilla interactions
- [x] Remove now empty variable/flag usages

### Government tweaks
- [x] Add "tributary" contract
- [x] Non-bureaucratic lieges of gubernatorials should change government to subroman
- [x] Fix independent gubernatorial_government upon independence; try to trigger "when landed" conditions to have can_get_government to be used instead of other code.
- [x] Refactor title_on_actions to suppress "trying to set illegal government" errors.

### New Features
- [x] Overhaul heresy event to support new faiths/religions
- [x] Overhaul character repopulation to fix ahistorical culture/religion spawns
- [x] Create Tributary Contracts
- [x] Tributary Wars
- [x] Offer Foederatus Interaction
- [x] Create decision to allow AI Pentarchs to seize baronies from co-religionsists when unlanded

### UI
- [x] Add new reformed icons to data set
- [x] Fix missing localizations for new holy sites
- [x] Fix missing localizations in foederatus offer
   * Namely needed for can_be_picked_title triggers
   * ~Also fixed screwed up promote is letter for foederatus~

### Gameplay Errors
- [x] Fix non-HOF theocracy religious titles being inheritable or revocable by secular rulers (H/T Abian36 )
  * Demanding conversion of Ecumenical Patriarch and Patriarchy of Antioch changes the inheritance laws.
  * Demanding theocrat conversion seems to invoke government fallback mechanics (make merc companies primary fallback government and using `set_faith` on patriarch of Jerusalem yields merc-company governments)
- [x] Potential CTD involved with some characters dying (namely patriarch of Jerusalem, Antioch) ~~[No longer seems to be happening]~~
   * Found cause of bug: common/on_action/dynasty_on_action.txt:5 will break when landed low borns (i.e., baron, counts) die. Some of the patriarchs are lowborn at start.
   * Either that, or its an issue when it occurs on day 1; marking as fixed for now
- [x] Offering a foederatus contract yields a script-system error when applying the foederatus contract in `on_accept`
- [x] Fix a few minor bugs associated with #67 (false conversion flags not propagating effects)
- [x] Fix Marzban contract issues

### Known Issues
* A small number of errors associated with current title_on_action and game_start government code; small enough to be pushed off for later
* Still some minor "illegal_government" errors when handing out titles, but significantly curtailed from previous examples.
* Fix women being able to inherit male-only titles in the 476 start
   * Seems to be an issue in Vanilla as well (see comment below)
* Landless, non-HOF theocracies are currently broken and should probably be removed _in toto_ and made a beta focus
   * Likely a conflict between not owning a county and lease mechanics
   * AI Pentarchs doesn't seem to be firing decision to retake baronies

### CODE REVIEW

**Comment by kai229** [comment]
[feudalism.log](https://github.com/loup99/BP/files/7378073/feudalism.log) Uploaded the requested error.log where Feudalism is appearing earlier than expected.

**Comment by LT-Rascek** [comment]
Reviewing the "Female successors at start" this seems to be a much deeper issue as I've seen this occur in Vanilla as well.

Example from a Vanilla, mod-less 1066 game I tested, a few days after letting the game run (Salic-law France).



I recommend shelving fixing this bug and reviewing Bug Reports (BR) to PDX. If no extant BRs cover this issue, then I will make one and send it to them.

**Comment by LT-Rascek** [comment]
If removing pentarchs is desired, I would recommend holding off on merging and letting me prune the related code before merge.

File: BP/common/casus_belli_types/00_vassalization.txt
Comment (loup99) [inline BP/common/casus_belli_types/00_vassalization.txt:1]: Do CBs in the same group have to be in the same file, or could the tributary CB be separate?

File: BP/common/character_interactions/00_vassal_interactions.txt
Comment (loup99) [inline BP/common/character_interactions/00_vassal_interactions.txt:1]: Could the exact change in this file be commented?

File: BP/common/culture/cultures/_BP.info
Comment (loup99) [inline BP/common/culture/cultures/_BP.info:46]: This is still an issue for your information

File: BP/common/decisions/10_religious_decisions.txt
Comment (loup99) [inline BP/common/decisions/10_religious_decisions.txt:1]: In the commits it is indicated that this is being added to debug Jomini, is that all or is there anything else here? Because I'm not convinced of the utility of adding files to debug vanilla, that should rather be a bug report.

File: BP/common/scripted_modifiers/00_building_modifiers.txt
Comment (loup99) [inline BP/common/scripted_modifiers/00_building_modifiers.txt:1]: Isn't this the exact same as in vanilla? Do we need to include it?

Comment (LT-Rascek) [reply BP/common/scripted_modifiers/00_building_modifiers.txt]: It's not; I probably had to tweak something that got overtaken by events; I've listed a full code cleanup as one of my last tasks for #71, but it's probably best done after we have an alpha candidate. Fixed on this branch though.

File: BP/common/decisions/10_religious_decisions.txt
Comment (LT-Rascek) [reply BP/common/decisions/10_religious_decisions.txt:1]: Sorry that commit message was a bit ambiguous; I had to suppress a vanilla flag (cannot_be_atoned_krstjani_leader_flag) that was throwing Jomini errors. I've highlighted the exact changes in the file.

File: BP/common/casus_belli_types/00_vassalization.txt
Comment (LT-Rascek) [reply BP/common/casus_belli_types/00_vassalization.txt]: They do not have to be the same; probably had some issue with it earlier and forgot to split it.

File: BP/common/scripted_effects/00_scripted_effects.txt
Comment (loup99) [inline BP/common/scripted_effects/00_scripted_effects.txt:1]: It is unclear what you changed in this file?

Comment (LT-Rascek) [reply BP/common/scripted_effects/00_scripted_effects.txt]: Nothing apparently; another holdover file.

**Review by loup99 — APPROVED** [review]

**Pull request merged**
